### DISCUSSION TITLE
Skipped tests have different phases depending on if skip is implemented with pytest.mark.skip decorator, or skipped in the test with pytest.skip

### DISCUSSION THREAD

**Discussion by jeffwright13** [pytest-dev/pytest#9814] [Q&A] [answered]
I have two different tests that implement skipping in different ways:
```
def test_d_skip():
    pytest.skip("skipping this test")
```
```
@pytest.mark.skip
def test_5_marked_SKIP(log_testname):
    assert 1
```

What I notice is that only the  'setup' and 'teardown' phases exist if a test is skipped with the `pytest.mark.skip` decorator; but all three phases exist ('setup', 'call' and 'teardown') if it is skipped with the `pytest.skip` statement inside the test.

Is there any specific reason for this difference in behavior?

**Comment by symonk** [comment]
I guess in order to hit the skip the test item is actually executed so having a 'call' phase seems alright for that use case, a test could fail before the skip for example and only skip under some logic.

**Reply by jeffwright13** [reply]
Yeah but wouldn't the result be the opposite from what I see? The decorator could be used by Pytest as a way to know that it doesn't even need to execute the test, so the call phase could be skipped. Whereas the test case which calls `pytest.skip` inside the test definition would need to be executed (and hence have a `call` phase) before it knew to skip.

**Comment by RonnyPfannschmidt** [comment, marked as answer]
It's behaviour as expected, the call phase only happens when the setup passes

**Reply by jeffwright13** [reply]
Thanks for the response. Still don't quite get it but the good news is, this forced me to rethink how tests are classified in my own code, and I just realized that instead of trying to recreate the complex logic of Pytest in my own after-test results processing, I can just leverage the `-v` option and categorize test results on how Pytest marks them in the verbose output. And that way if anything changes in the future I won't have to redo my code.
